### ISSUE TITLE
Методы для добавления новой недвижимости в БД

### ISSUE THREAD

**Issue by eastwing** [rgrdfo/rgr.core#13]
Необходимо разработать алгоритмы для добавления новых объектов недвижимости в БД, автоматического преобразования добавленных файлов, и т.д.

**Comment by PKoroboff** [comment]
Алгоритмы разные в зависимости от типа объекта недвиимости и операции. Сейчас работаем только над продажей и в первую очередь квартиры, комнаты, дома, земельные участки и др.
Общая логика такая:
Когда пользователь добавляет новый объект он автоматически сохраняется в статус черновик и виден только этому пользователю, никто кроме него этот объект не видит.
Когда пользователь хочет показать этот объект другим (то есть задействовать весь функционал, который возможен в системе) - он меняет его статус на Активный (как сейчас это делается), а по новому - Публикует. Когда пользователь нажимает на кнопку Опубликовать запускается алгоритм (он сейчас реализован и работает) при смене статуса на Активный (во вложении)
![_ 09 0814](https://cloud.githubusercontent.com/assets/3958508/20164172/9d782ab6-a756-11e6-9cb1-b1fd6819ebb4.png)

Этот алгоритм проверяет уникальность объекта, чтобы не было дублей, а так же проверяет тип договора у добавляемого объекта

**Comment by PKoroboff** [comment]
При указании стоимости объекта в раздел продажа сейчас используется вот этот алгоритм
![default](https://cloud.githubusercontent.com/assets/3958508/20164073/c31bb1c6-a755-11e6-9827-0f7b29f74937.png)
Этот алгоритм запускается по выходу из поля Стоимость. Он проверяет, чтобы цена указывалась в рублях, а не в тысячах рублей.

**Comment by PKoroboff** [comment]
Вот с этим мне непонятно, поясни

> автоматического преобразования добавленных файлов, и т.д.

**Comment by eastwing** [comment]
Добавденый файл загружается на сервер, переименовывается (при этом в БД записывается его исходное имя). Если файл хранится физически на сервере, он помещается в две таблицы, таблицу ссылок и таблицу хранилища. Плюс сейчас хотят добавлять видео и какой-то 3д-тур, которые мы на своём сервере хранить определённо не сможем. В этом случае надо сделать запись в таблицу ссылок.

Вот это всё я имел в виду. Если вкратце.

**Comment by PKoroboff** [comment]
Что еще за "файл"? У нас из файлов только файлы фотографий пока есть, больше никаких нет. Ты о фото?

**Comment by PKoroboff** [comment]
Видео добавляться будет через ссылку на ютуб, а про 3д тур я вообще в первый раз слышу. Когда ты говоришь 

> хотят добавлять видео и какой-то 3д-тур

ты кого имеешь в виду?

**Comment by eastwing** [comment]
"Файл" - это именованный структурированный набор байтов. Например, фотография. Делать надо сразу универсально, если появятся какие-то другие "файлы".

Добавление ссылки на ютуб - это так же процесс слздания записи в БД, о чём я уже написал.

3д-тур - это примерно так же

**Comment by PKoroboff** [comment]
Согласен, это ты про блоб-поля имеешь в виду? Разве сейчас не так реализовано?

**Comment by eastwing** [comment]
Я не знаю, что такое блоб-поле. Имеется в виду вот [это](http://rbn.estate/3D/kv/Shevchenko4.html). Это блоб-поле? Если да, то я даже не знал, что они сейчас есть.

Об этих 3д-турах в последние дни столько говорят, что я думаю, их добавление неизбежно

**Comment by PKoroboff** [comment]
блоб-поля - это именованный структурированный набор байтов.

Пусть говорят сколько угодно. У нас есть внутренний регламент работы, по которому мы согласовываем те фичи, которые берем в разработку на ближайший период. Поскольку я про добавление 3д-туров слышу впервые, значит мы это не обсуждали.
Я понимаю кто может педалировать эти туры. Критерии установки приоритетов задач из бэклога определяем мы, а не конкретные персоны

**Comment by eastwing** [comment]
> Я понимаю кто может педалировать эти туры

Не только, ещё руководитель комитета по ИТ
